src/context/src/input.rs

```diff
@@ -31,4 +38,14 @@ impl InputHandler {
     pub fn key_down(&self, key: VirtualKeyCode) -> bool {
         self.keys_down.contains(&key)
     }
-}
\ No newline at end of file
+
+    ///Check if `button` is pressed.
+    pub fn button_down(&self, button: MouseButton) -> bool {
```

Comment:
I'm not sure. clicked implies that the button wasn't clicked in the previous logic frame, fn `mouse_key_down` would work.

Maybe we should rethink fn `key_down`. It could handle keyboard and mouse buttons. That would mean that it would take something like `Either<T>` where Either<T>::get() returns Self::Type. I'm not certain though, generics are hard to do right.
I think we can start with `fn mouse_key_down` and `fn keyboard_key_down` and then extend it later.